lib/sles4sap_publiccloud.pm

```diff
@@ -1240,9 +1240,10 @@ sub pacemaker_version {
 
 sub saphanasr_showAttr_version {
     my ($self) = @_;
-    my $version_cmd = 'SAPHanaSR-showAttr --version';
 
-    my $version_output = $self->run_cmd(cmd => $version_cmd, quiet => 1);
+    my $which_output = $self->run_cmd(cmd => 'which SAPHanaSR-showAttr', quiet => 1);
+    my $version_output = $self->run_cmd(cmd => "rpm -qf $which_output", quiet => 1);
```

Comment:
We can merge these two lines as:
```
    my $version_output = $self->run_cmd(cmd => 'rpm -qf $(which SAPHanaSR-showAttr)', quiet => 1);
```

Reply:
> We can merge these two lines as:
> 
> ```
>     my $version_output = $self->run_cmd(cmd => 'rpm -qf $(which SAPHanaSR-showAttr)', quiet => 1);
> ```

My original test code was merging these two lines into one as you suggested. But test cases failed on something like:
```
which: no SAPHanaSR-showAttr in $PATH
rpm: no arguments gien for query
```
Then I tried `export PATH=$PATH:/usr/sbin` but it failed on:
` sudo: export: command not found`

Then I decided to separate it to two lines.

See the pic attached for more details.

Reply:
![Uploading Screenshot from 2025-02-08 16-47-31.png…]()

Reply:
I think I see the issue. `$(which SAPHanaSR-showAttr)` runs first on the openQA VM (the jumphost), before sending the command to the SUT in the cloud. So in the end, I guess it was not easily possible to combine the 2 commands. Only possibility I can think of would be doing something like:
```
    my $version_output = $self->run_cmd(cmd => 'rpm -qf \$(which SAPHanaSR-showAttr)', quiet => 1);
```
But even then, I'm not sure it'll work. In the end you went with the best approach. Thanks!